setup.py

```diff
@@ -5,30 +5,25 @@
 from setuptools import setup
 import versioneer
 
-# NOTE: These are tested in `continuous_integration/travis/test_imports.sh` If
+# NOTE: These are tested in `continuous_integration/test_imports.sh` If
 # you modify these, make sure to change the corresponding line there.
 extras_require = {
-    "array": ["numpy >= 1.15.1", "toolz >= 0.8.2"],
-    "bag": [
-        "cloudpickle >= 0.2.2",
-        "fsspec >= 0.6.0",
-        "toolz >= 0.8.2",
-        "partd >= 0.3.10",
-    ],
-    "dataframe": [
-        "numpy >= 1.15.1",
-        "pandas >= 0.25.0",
-        "toolz >= 0.8.2",
-        "partd >= 0.3.10",
-        "fsspec >= 0.6.0",
-    ],
+    "array": ["numpy >= 1.15.1"],
+    "bag": [],  # keeping for backwards compatibility
+    "dataframe": ["numpy >= 1.15.1", "pandas >= 0.25.0"],
```

Comment:
No need to do anything in this PR, but I'm curious when do we want to bump these and what versions should we bump them to?

Reply:
We bumped them a few months ago to this. There was some talk in the maintenance meeting about using released > n months ago as the floor. Apparently xarray does something like this? But we didn't choose a specific policy

Reply:
At xarray and pint we use a rolling policy based on NEP-29:
https://xarray.pydata.org/en/stable/installing.html#minimum-dependency-versions
The number of months listed is arbitrary and subject to negotiation. The key benefit of such a policy is that it allows developer to bump up minimum dependencies as needed, without initiating a discussion every time.

Reply:
Leaving alone is fine. I think we have some workarounds for NumPy pre-1.17 that could be cleaned out once that is a minimum

cc @pentschev (in case you have thoughts here 🙂)

Reply:
Yeah, I also agree we could consider dropping older versions of NumPy. As @crusaderky mentioned NEP-29, it's worth noticing that NumPy is supporting Python releases for 18 months. Is there any reason for us to keep on supporting NumPy versions (or all other dependencies for that matter) that are much older than that?

Reply:
I opened https://github.com/dask/dask/issues/7378 to continue this conversation